cpp/src/parquet/file_writer.cc

```diff
@@ -335,6 +354,7 @@ class FileSerializer : public ParquetFileWriter::Contents {
       row_group_writer_.reset();
 
       WritePageIndex();
+      WriteBloomFilter();
```

Comment:
According to the [specs](https://github.com/apache/parquet-format/blob/master/BloomFilter.md),
- The Bloom filter data can be stored before the page indexes after all row groups.
- Or it can be stored between row groups

So `WriteBloomFilter()` need to be placed above `WritePageIndex()`

Reply:
BTW, if they are stored between row groups, it would help relax memory pressure.

Reply:
May I ask that why bloom filter data should before PageIndex?